tests/helper/tests/debian_cis.py

```diff
@@ -1,98 +1,63 @@
-            config_src = feature_config[my_feature]["config_src"]
-            config_dst = feature_config[my_feature]["config_dst"]
-            script_src = feature_config[my_feature]["script_src"]
-            script_dst = feature_config[my_feature]["script_dst"]
-
-            # /tmp has 'noexec' flag; therefore we need to
-            # call each script with bash.
-            cmd_debian_cis = "for i in `ls /tmp/debian-cis/bin/hardening/*.sh`; do /bin/bash $i; done"
-
-            (exit_code, output, error) = client.execute_command(f"cd /tmp/ && git clone {git_debian_cis}")
-            assert exit_code == 0, f"no {error=} expected"
-
-            (exit_code, output, error) = client.execute_command(f"cd /tmp/ && git clone {git_gardenlinux}")
-            assert exit_code == 0, f"no {error=} expected"
-
-            (exit_code, output, error) = client.execute_command("echo CIS_ROOT_DIR='/tmp/debian-cis' > /etc/default/cis-hardening")
-            assert exit_code == 0, f"no {error=} expected"
-
-            (exit_code, output, error) = client.execute_command(f"cp {config_src} {config_dst}")
-            assert exit_code == 0, f"no {error=} expected"
-
-
-            (exit_code, output, error) = client.execute_command(f"cp {script_src} {script_dst}")
-            assert exit_code == 0, f"no {error=} expected"
-
-            (exit_code, output, error) = client.execute_command(f"{cmd_debian_cis}")
-            if "Check Failed" in output:
-                exit_code == 1
-                assert output
-            assert exit_code == 0, f"no {error=} expected"
-
-        return cls.instance
+def DebianCIS(client, config, args):
+    """Performing unit test for feature: CIS"""
+
+    # Log used CIS test config
+    logger.info(f"CIS config: {config}")
+
+    # Defining vars from PyTest parametrize
+    git_debian_cis =  args["git_debian_cis"]
+    git_debian_cis_branch = args["git_debian_cis_branch"]
+    git_gardenlinux = args["git_gardenlinux"]
+    git_gardenlinux_branch = args["git_gardenlinux_branch"]
+    config_src = args["config_src"]
+    config_dst = args["config_dst"]
+    script_src = args["script_src"]
+    script_dst = args["script_dst"]
+
+    # /tmp has 'noexec' flag; therefore we need to
+    # call each script with bash.
+    cmd_debian_cis = "for i in `ls /tmp/debian-cis/bin/hardening/*.sh`; do /bin/bash $i; done"
+
+    # Fetch DebianCIS GIT repository from upstream
+    (exit_code, output, error) = client.execute_command(
+        f"cd /tmp/ && git clone {git_debian_cis}")
+    assert exit_code == 0, f"no {error=} expected"
+
+    # Checkout defined branch for DebianCIS GIT
+    (exit_code, output, error) = client.execute_command(
+        f"cd /tmp/debian-cis && git checkout {git_debian_cis_branch}")
+    assert exit_code == 0, f"no {error=} expected"
+
+    # Fetch Garden Linux repository from upstream
+    (exit_code, output, error) = client.execute_command(
+        f"cd /tmp/ && git clone {git_gardenlinux}")
```

Comment:
Instead of cloning the Garden Linux Git repo, we should probably just copy the required directories (`conf.d` and `check_scripts`) into the container via `bulk_upload`